### DISCUSSION TITLE
[@types/opentype.js] needs updating for current release.

### DISCUSSION THREAD

**Discussion by ILOVEPIE** [DefinitelyTyped/DefinitelyTyped#64229] [Issues with a @types package]
I'm one of the maintainers for opentype.js.

Just to let you guys know, there's a breaking change coming in the next major version with regard to the ``Font.names`` field.
The way opentype.js was handling names wouldn't properly keep track of platform specific names so we're implementing a change. You will need to create a new interface that contains variables ``unicode``,``macintosh``, and ``windows`` which will be either undefined or of type ``FontNames``.
The ``Font.names`` field will need to be changed to the new type.

Not only will this fix platform specific names for generated fonts, but it also will allow users to specify them or use names only from a specific platform.

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
@ILOVEPIE thanks for the head-ups!
If the major version would be used (2.*.*) there won't be issue for library users with TS implementation. The TS tools (cli, VSCode, etc) handle version resolutions nicely. The change would have to be done here for `@types/opentype.js` ofc with version bump

**Reply by ILOVEPIE** [reply]
The 2.0.0 release was released, we didn't get everything we wanted in it, so there may be additional functionality coming.

**Reply by ILOVEPIE** [reply]
also i found an old typing issue that hasn't been addressed in your discussions: #69680

**Reply by kshitijanurag** [reply]
> also i found an old typing issue that hasn't been addressed in your discussions: #69680

Opened https://github.com/DefinitelyTyped/DefinitelyTyped/pull/75056 to fix the typing issue.
